test/integration_test.py

```diff
@@ -1,18 +1,30 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
 
+import argparse
 import subprocess
 import sys
-import argparse
+import os
 import re
+import shutil
 import threading
+import tempfile
+
+
+# Set up the file structure
+root_dir = tempfile.mkdtemp(prefix='mock_contents')
+os.mkdir(os.path.join(root_dir, 'src'))
+with open(os.path.join(root_dir, 'src', 'temp.txt'), 'w') as fid:
+    fid.write('hello')
+
 
 KARMA_PORT = 9876
 
 
 def start_notebook():
     nb_command = [sys.executable, '-m', 'notebook', '--no-browser',
-                  '--debug', '--NotebookApp.allow_origin="*"']
+                  '--debug', '--NotebookApp.allow_origin="*"',
+                  '--NotebookApp.notebook_dir="%s"' % root_dir]
```

Comment:
Minor detail: notebook_dir is already a positional arg, so the `notebook_dir=` can be left out, leaving Popen to deal with quoting/escaping root_dir.